libraries/Microsoft.Bot.Builder.Runtime/Extensions/IConfigurationBuilderExtensions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.Runtime.Extensions
+{
+    // TODO #22: Refactor bot path adapter
+
+    /// <summary>
+    /// Bot path adapter, for development environment, use '../../' as the bot path, for deployment and production environment, use 'ComposerDialogs' as bot path.
+    /// </summary>
+    public static class IConfigurationBuilderExtensions
+    {
+        private const string ComposerDialogsDirectoryName = "ComposerDialogs";
+        private const string DevelopmentApplicationRoot = "../../../";
```

Comment:
should this path be configurable?

Reply:
I don't think there really is a strong need that we've identified to need to configure the path. The main reason for this really is due to the nature in which Composer structures the project when it is written to disk, and also in how they orchestrate the copying and moving of files when generating builds for publishing to Azure.

In an ideal world, we should always just assume the current operating directory, which would work when working at source code project root level (since assets would be available from that level as well as code - currently, they are above the project directory per what Composer does) or from the root of the publish folder for deployments.

One of the goals for R12 is to fix the project structure to address this, which should then really remove the need for this, so ideally a lot of the complexity here goes away. Where I could see configuration maybe being applicable is specifying sources for assets to include that may be non-standard (e.g. remote store of assets). Since the SDK right now I believe only supports local folders for declarative assets, then I figure we can address that down the line.